extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/tracing/MongoTracingCommandListener.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.mongodb.tracing;
+
+import com.mongodb.event.CommandFailedEvent;
+import com.mongodb.event.CommandListener;
+import com.mongodb.event.CommandStartedEvent;
+import com.mongodb.event.CommandSucceededEvent;
+
+import io.opentracing.contrib.mongo.common.TracingCommandListener;
+import io.opentracing.util.GlobalTracer;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+/**
+ * Command Listener for Mongo client delegated to {@link TracingCommandListener}.
+ * 
+ */
+public class MongoTracingCommandListener implements CommandListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoTracingCommandListener.class);
+
+    private TracingCommandListener delegate;
+
+    public MongoTracingCommandListener() {
+        this.delegate = new TracingCommandListener.Builder(GlobalTracer.get()).build();
```

Comment:
If this is the only reason why we cannot directly use the `TracingCommandListener` maybe open an issue upstream (and propose a PR) to avoid maintaining this `MongoTracingCommandListener` that only build the `TracingCommandListener` with the GlobalTracer.

You can keep this class while waiting for the fix upstream as a workaround